pkg/packagekit/updates.jsx

```diff
@@ -190,36 +193,69 @@ function HeaderBar(props) {
     );
 }
 
-function UpdateItem(props) {
-    const info = props.info;
-    var bugs = null;
-    var security_info = null;
-
-    if (info.bug_urls && info.bug_urls.length) {
-        // we assume a bug URL ends with a number; if not, show the complete URL
-        bugs = commaJoin(info.bug_urls.map(u => <a rel="noopener" referrerpolicy="no-referrer" target="_blank" href={u} >{u.match(/[0-9]+$/) || u}</a>));
+class UpdateItem extends React.Component {
+    constructor() {
+        super();
+        this.state = {expanded: false};
     }
 
-    if (info.security) {
-        security_info = (
-            <p>
-                <span className="fa fa-bug security-label"> </span>
-                <span className="security-label-text">{ _("Security Update") + (info.cve_urls.length ? ": " : "") }</span>
-                { commaJoin(info.cve_urls.map(u => <a href={u} rel="noopener" referrerpolicy="no-referrer" target="_blank">{u.match(/[^/=]+$/)}</a>)) }
-            </p>
+    render() {
+        const info = this.props.info;
+        var bugs = null;
+        var security_info = null;
+
+        if (info.bug_urls && info.bug_urls.length) {
+            // we assume a bug URL ends with a number; if not, show the complete URL
+            bugs = commaJoin(info.bug_urls.map(u => <a rel="noopener" referrerpolicy="no-referrer" target="_blank" href={u} >{u.match(/[0-9]+$/) || u}</a>));
+        }
+
+        if (info.security) {
+            security_info = (
+                <p>
+                    <span className="fa fa-bug security-label"> </span>
+                    <span className="security-label-text">{ _("Security Update") + (info.cve_urls.length ? ": " : "") }</span>
+                    { commaJoin(info.cve_urls.map(u => <a href={u} rel="noopener" referrerpolicy="no-referrer" target="_blank">{u.match(/[^/=]+$/)}</a>)) }
+                </p>
+            );
+        }
+
+        /* truncate long package list by default */
+        var pkgList = this.props.pkgNames.map(n => (<Tooltip tip={packageSummaries[n]}><span>{n}</span></Tooltip>));
+        var pkgs;
+        if (!this.state.expanded && pkgList.length > 15) {
+            pkgs = (
+                <div onClick={ () => this.setState({expanded: true}) }>
+                    {commaJoin(pkgList.slice(0, 15))}
+                    <a className="info-expander">{ cockpit.format(_("$0 more…"), pkgList.length - 15) }</a>
+                </div>);
+        } else {
+            pkgs = commaJoin(pkgList);
+        }
+
+        /* truncate long description by default */
+        var descLines = (info.description || "").trim().split("\n");
+        var desc;
+        if (!this.state.expanded && descLines.length > 7) {
+            desc = (
+                <div onClick={ () => this.setState({expanded: true}) }>
```

Comment:
It looks like clicking to expand either the list of packages or the description will expand both... is that intended? See comment in the tests also, regarding this.